trace-utils/src/msgpack_decoder/decode/meta_struct.rs

```diff
@@ -3,33 +3,64 @@
 
 use crate::msgpack_decoder::decode::error::DecodeError;
 use crate::msgpack_decoder::decode::map::{read_map, read_map_len};
-use crate::msgpack_decoder::decode::number::read_number_bytes;
 use crate::msgpack_decoder::decode::string::{handle_null_marker, read_string_bytes};
 use rmp::decode;
 use std::collections::HashMap;
 use tinybytes::{Bytes, BytesString};
 
+fn read_byte_array_len(buf: &mut &[u8]) -> Result<u32, DecodeError> {
```

Comment:
It is my understanding that if you use `#[inline]` on a public function you also want to use it on private functions that are called from within the public one. 

That being said, I'm not 100% sure how we're using `#[inline]` right now in the decoder is necessary anyway. But we can discuss that separately.